pkg/controller/config/config_controller.go

```diff
@@ -0,0 +1,349 @@
+	)
+}
+
+// Add creates a new Config Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	m, err := mf.NewManifest(resourceDir, recursive, mgr.GetClient())
+	if err != nil {
+		return err
+	}
+	return add(mgr, newReconciler(mgr, m))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, m mf.Manifest) reconcile.Reconciler {
+	return &ReconcileConfig{
+		client:   mgr.GetClient(),
+		scheme:   mgr.GetScheme(),
+		manifest: m,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	log := ctrlLog.WithValues("context", "watch")
+	// Create a new controller
+	c, err := controller.New("config-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Config
+	log.Info("Watching operator config CR")
+	err = c.Watch(
+		&source.Kind{Type: &op.Config{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.GenerationChangedPredicate{},
+	)
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(
+		&source.Kind{Type: &appsv1.Deployment{}},
+		&handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &op.Config{},
+		})
+	if err != nil {
+		return err
+	}
+
+	if noAutoInstall {
+		return nil
+	}
+
+	if err := createCR(mgr.GetClient()); err != nil {
+		log.Error(err, "creation of config resource failed")
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileConfig implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileConfig{}
+
+// ReconcileConfig reconciles a Config object
+type ReconcileConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client   client.Client
+	scheme   *runtime.Scheme
+	manifest mf.Manifest
+}
+
+// Reconcile reads that state of the cluster for a Config object and makes changes based on the state read
+// and what is in the Config.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileConfig) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	log := requestLogger(req, "reconcile")
+
+	log.Info("reconciling config chnage")
+
+	// NOTE: only interested in the clusterwide events of the resource watched
+	// Otherwise, when release.yaml is applied and onwer-ref is set to the
+	// clusterwide resource, events are generated for "targetNamespace/resourceWatched"
+	// and as the GET call for that resource fails, the pipeline resources get
+	// deleted. this filter prevents that from happening
+	if req.Name == resourceWatched && req.Namespace != "" {
```

Comment:
shall we move `if req.Name == resourceWatched && req.Namespace != ""` this into a function
The big comment ruins the flow while I read the code

something like 

```
func acceptEvent(r req reconcile.Request) bool 
```

Reply:
I think a better way is to write the `predicate`. I will address that in a different pr but `acceptEvent` sounds fine.